### PR TITLE
drivers: sfc: Code implements for read/erase/write flash on xip.

### PR DESCRIPTION
**Pull request by zbjlu**
Code implements for read/erase/write flash on xip.

Signed-off-by: Stephen.Zhang <Stephen.Zhang@spreadtrum.com>

### CODE REVIEW

File: drivers/src/hal_sfc.c [hunk 1]
```diff
@@ -531,12 +519,15 @@ void spiflash_cmd_write(struct spi_flash *flash, const u8_t * cmd,
 	spiflash_read_write(cmd_desc, cmd_length, NULL);
 }
 
-int spiflash_cmd_poll_bit(struct spi_flash *flash, unsigned long timeout,
-		u8_t cmd, u32_t poll_bit, u32_t bit_value)
+__ramfunc int spiflash_cmd_poll_bit(struct spi_flash *flash,
+	unsigned long timeout, u8_t cmd, u32_t poll_bit, u32_t bit_value)
 {
 	u32_t status = 0;
+	u32_t count	 = 300000;
```
Comment (easonxiang) [inline drivers/src/hal_sfc.c]: Can you add some comments to describe what does this number mean?

File: drivers/src/hal_sfc.c [hunk 2]
```diff
@@ -1129,7 +1152,7 @@ int spiflash_read_data_xip(struct spi_flash *flash, u32_t offset,
 	spiflash_set_xip_cmd(flash, (const u8_t *)(&off_cmd), dump_byte);
 	*buf = spiflash_read_common(flash, offset);
 
-	return 0;
+	return TRUE;
```
Comment (easonxiang) [inline drivers/src/hal_sfc.c]: Any reason to change this?

File: drivers/src/hal_sfc.c [hunk 3]
```diff
@@ -1266,7 +1289,34 @@ static int spiflash_set_qpi(struct spi_flash *flash, u32_t op)
...
+	memcpy((char *)buf, (char *)(offset), len);
+#endif
+
+	return ret;
+}
+
+__ramfunc static void spiflash_exit_xip(void)
 {
 	/* pin mux and drive configuration */
 	sci_write32(0x40840038, 0x4100);
```
Comment (easonxiang) [inline drivers/src/hal_sfc.c]: Can you define all registers and value with macro, please look at drivers/src/hal_pinmux.c for reference.

File: drivers/src/hal_sfc.c [hunk 4]
```diff
@@ -923,10 +942,13 @@ int spiflash_cmd_program(struct spi_flash *flash, u32_t offset,
 	return ret;
 }
 
-int spiflash_write_enable(struct spi_flash *flash)
+__ramfunc int spiflash_write_enable(struct spi_flash *flash)
 {
-#if 0
+#if 1
```
Comment (easonxiang) [inline drivers/src/hal_sfc.c]: unused code?

File: drivers/src/hal_sfc.c (hunk 1 above)
Comment (zbjlu) [reply drivers/src/hal_sfc.c]: fixed

**Pull request merged**